packages/embark/src/lib/utils/template_generator.js

```diff
@@ -73,52 +74,150 @@ class TemplateGenerator {
+      templateSpecifier = `${templatePkg}@^${version}`;
+    }
+
+    const tmpDir = require('fs-extra').mkdtempSync(
+      utils.joinPath(require('os').tmpdir(), `${this.templateName}-`)
+    );
+
+    execSync(`npm pack ${templateSpecifier}`, {cwd: tmpDir, stdio: 'ignore'});
+    const packed = utils.filesMatchingPattern(
+      [utils.joinPath(tmpDir, '*.tgz')]
+    )[0];
+
+    this.extract(packed, fspath, () => {
+      this.installTemplate(
+        fspath,
+        name,
+        true,
+        () => {
+          if (name === 'embark_demo') {
+            console.log('-------------------'.yellow);
+            console.log(__('Next steps:').green);
+            console.log(('-> ' + ('cd ' + fspath).bold.cyan).green);
+            console.log('-> '.green + 'embark run'.bold.cyan);
+            console.log(__('For more info go to http://embark.status.im').green);
+          }
+        }
+      );
+    });
+  }
+
+  insideMonorepo() {
+    if (this._monorepoRoot === undefined) {
+      const maybeInsideMonorepo = fs.existsSync(
+        fs.embarkPath('../../packages/embark')
+      );
+      let monorepoRoot = null;
+      if (maybeInsideMonorepo) {
+        const lernaJsonPath = findUp.sync('lerna.json', {cwd: fs.embarkPath()});
+        if (lernaJsonPath) {
+          monorepoRoot = utils.dirname(lernaJsonPath);
         }
       }
-    );
+      this._monorepoRoot = monorepoRoot;
+    }
+    return this._monorepoRoot;
   }
 
   installTemplate(templatePath, name, installPackages, cb) {
     utils.cd(templatePath);
-    utils.sed('package.json', '%APP_NAME%', name);
+
+    const pkgJson = fs.readJSONSync('./package.json');
+    pkgJson.name = name;
+    if (!(/demo/).test(name)) {
+      pkgJson.version = '0.0.1';
+    }
+    delete pkgJson.files;
+    if (!pkgJson.scripts) pkgJson.scripts = {};
+    delete pkgJson.scripts.ci;
+    delete pkgJson.scripts.clean;
+    delete pkgJson.scripts.package;
+    delete pkgJson.scripts.qa;
+    delete pkgJson.scripts.reset;
+    pkgJson.scripts.test = 'embark test';
+    if (pkgJson.devDependencies) {
+      delete pkgJson.devDependencies['embark'];
+      delete pkgJson.devDependencies['embark-reset'];
+      delete pkgJson.devDependencies['npm-run-all'];
+      delete pkgJson.devDependencies['rimraf'];
+    }
+    fs.writeFileSync('package.json', JSON.stringify(pkgJson, null, 2));
+
     if (fs.existsSync('dot.gitignore')) {
-      fs.moveSync('dot.gitignore', '.gitignore');
-    } else if (!fs.existsSync('.gitignore')) {
-      fs.copySync(fs.embarkPath('templates/dot.gitignore'), '.gitignore');
+      fs.moveSync('dot.gitignore', '.gitignore', {overwrite: true});
+    }
+    if (fs.existsSync('dot.npmrc')) {
+      fs.moveSync('dot.npmrc', '.npmrc', {overwrite: true});
     }
 
     if (installPackages) {
       console.log(__('Installing packages...').green);
+      const monorepoRoot = this.insideMonorepo();
+      if (monorepoRoot) {
```

Comment:
Using the suggestion above, this would turn in to
```
if (this.monorepoRootPath) {
  // ...
  execSync(
    'npx lerna list --long --parseable',
     {cwd: this.monorepoRootPath,
      stdio: ['ignore', 'pipe', 'ignore']}
  )
}
```

Reply:
👍